cmd/spire-agent/main.go

```diff
@@ -4,8 +4,9 @@ import (
 	"os"
 
 	"github.com/spiffe/spire/cmd/spire-agent/cli"
+	"github.com/spiffe/spire/pkg/common/entrypoint"
 )
 
 func main() {
-	os.Exit(new(cli.CLI).Run(os.Args[1:]))
+	os.Exit(entrypoint.NewEntryPoint(new(cli.CLI).Run).Main())
```

Comment:
We discussed this and we agreed that in order to avoid blocking SPIRE to run only because the `svc.IsWindowsService()` fails, we will fallback to run SPIRE as a regular process if `IsWindowsService()` returns an error.